### ISSUE TITLE
IconGroupEntry.replaceIconsForResource() RangeError: Invalid typed array length

### ISSUE THREAD

**Issue by xBiggs** [jet2jet/resedit-js#30]
I got an error when trying to change the icon for an executable generated by [pkg](https://github.com/vercel/pkg).

```
/node_modules/resedit/dist/util/functions.js:43
        : new Uint8Array(src, srcOffset, length);

RangeError: Invalid typed array length: 1920
    at new Uint8Array (<anonymous>)
    at Object.copyBuffer (/node_modules/resedit/dist/util/functions.js:43:11)
    at IconItem.generate (/node_modules/resedit/dist/data/IconItem.js:149:21)
    at /node_modules/resedit/dist/resource/IconGroupEntry.js:176:31
    at Array.map (<anonymous>)
    at IconGroupEntry.replaceIconsForResource (/node_modules/resedit/dist/resource/IconGroupEntry.js:155:34)
    at file:///dist/winbuild.js:9:25

Node.js v18.4.0
```

The error was generated by this:

```ts
import { readFile, writeFile } from 'fs/promises';
import { join } from 'path';
import { fileURLToPath } from 'url';

import { NtExecutable, NtExecutableResource, Data, Resource } from 'resedit';

const __dirname = fileURLToPath(new URL('.', import.meta.url));

const main = async () => {
  try {
    const binary = await readFile(join(__dirname, '../pkg/win.exe'))
    const exe = NtExecutable.from(binary);
    const resource = NtExecutableResource.from(exe);

    const icon = await readFile(join(__dirname, 'pdf.ico'));
    const icon_file = Data.IconFile.from(icon);
    Resource.IconGroupEntry.replaceIconsForResource(resource.entries, 101, 1033, icon_file.icons.map(item => item.data));

    const vi_list = Resource.VersionInfo.fromEntries(resource.entries);
    const vi = vi_list.at(0);
    if (!vi) throw Error('vi is undefined');
    vi.setFileVersion(1, 0, 0, 0, 1033);
    vi.setStringValues(
      { lang: 1033, codepage: 1200 },
      {
        FileDescription: 'My application',
        ProductName: 'My product',
      },
    );
    vi.outputToResourceEntries(resource.entries);

    resource.outputResource(exe);
    const output = exe.generate();
    await writeFile(join(__dirname, '../pkg/sample.exe'), Buffer.from(output));
  } catch (e) {
    console.error((e as Error).message);
  }
}

main();
```

I have attached a sample executable and icon file.
[ExeIcon.zip](https://github.com/jet2jet/resedit-js/files/10187036/ExeIcon.zip)

Any help would be greatly appreciated.

**Comment by jet2jet** [comment]
Thanks for the report.  
I found the bug for width calculation and fixed it.

I published as v1.6.1 so please use this version and confirm whether the error does not occur.

**Comment by xBiggs** [comment]
I have confirmed your fix. Thanks!
